ocean_lib/web3_internal/contract_utils.py

```diff
@@ -59,6 +63,12 @@ def get_addresses_with_fallback(config):
 @enforce_types
 def get_contracts_addresses(config) -> Optional[Dict[str, str]]:
     """Get addresses for all contract names, per network and address_file given."""
+    if "CHAIN_ID" not in config:
+        w3 = get_web3(config["RPC_URL"])
+        # cache it to prevent further calls
+        # TODO: maybe we shouldn't?
```

Comment:
Now I saw that you have kept this key. In my PR I have removed all the occurrences.

Reply:
yes, because Alex asked us to save RPC calls, and therefore I'm caching the CHAIN_ID key. I think it is a reasonable compromise.